### PR TITLE
948  typeerror type string is not a subtype of type map string dynamic 

### PR DESCRIPTION
**Pull request by YassinNouh21**
📝 **Summary**
---
**This PR fixes**  #948 
--- 
### Description
- **Primary Problem**: The application encounters a type mismatch error when parsing HTTP requests in JSON format. The specific error message is: `type 'String' is not a subtype of type 'Map<String, dynamic>'`.
- **Error Context**: This error occurs in the `onItemError` callback. It is triggered when an exception is caught in a stream, such as `getMosqueTimesStream`.

### How the Issue is Triggered
- **Test Scenario**: The issue is reproduced by creating a dynamic fake string that fails to be correctly parsed by the `.fromMap` factory in the models.
- **Example Code**:
    ```dart
    static Future<Times> getMosqueTimes(String id) async {
      final response = await dio.get('/3.1/mosque/$id/times');
      final dynamic test = ‘test’;
      return Times.fromMap(test);
    }
    ```

### Solution Implemented
- **Approach**: The problem was resolved by implementing an interceptor in Dio (a powerful HTTP client for Dart).
- **Interceptor Functionality**: The interceptor checks if the response body is not in JSON format. If it's not, it throws an exception.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -109,6 +109,7 @@ dependencies:
   unique_identifier: ^0.3.0
   collection: any
   disk_space: ^0.2.1
+  mocktail: ^1.0.2
```
Comment (ghassenbenzahra123) [inline pubspec.yaml:112]: Please remove unused dependency

File: test/src/services/dio_interceptor.dart
Comment (ghassenbenzahra123) [inline test/src/services/dio_interceptor.dart:1]: Test files should always end with _test.dart

**Review by ghassenbenzahra123 — COMMENTED** [review]
Great work Yassin! 🚀

File: test/src/services/dio_interceptor.dart
Comment (YassinNouh21) [reply test/src/services/dio_interceptor.dart:1]: Ok I will change it

**Review by ghassenbenzahra123 — APPROVED** [review]

**Pull request merged**